### PR TITLE
FSR-594 add below zero message

### PR DESCRIPTION
**Pull request by cliff-defra**

### CODE REVIEW

File: test/models/river-and-sea-levels.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ lab.experiment('river-and-sea-levels model test', () => {
     })
 
     Code.expect(station.formattedValue).to.equal('0.22m')
-    Code.expect(rainfallStation.formattedValue).to.equal('0.0mm')
+    Code.expect(rainfallStation.formattedValue).to.equal('0m')
```
Comment (nikiwycherley) [inline test/models/river-and-sea-levels.js:64]: rainfall should be set 1 dp and shouldn't be effected by this ticket

File: server/views/station.html [hunk 2]
```diff
@@ -235,12 +235,16 @@ <h2 class="govuk-heading-m govuk-!-margin-top-6">How levels here could affect ne
       {% for band in model.thresholds %}
       <div class="defra-flood-impact-list__row {% if band.isLatest %} defra-flood-impact-list__row--current {% elif band.isExceeded %} defra-flood-impact-list__row--exceeded{% endif %}" {% if band.type === 'historical' %} data-toggle-list-display-item="impact" {% endif %}>
         <dt class="defra-flood-impact-list__key">
-          {{ band.level }}m
+          {% if not model.station.isCoastal and band.level <= 0 %}
```
Comment (nikiwycherley) [inline server/views/station.html:238]: logic should be worked out in the model

File: server/views/station.html [hunk 3]
```diff
@@ -235,12 +235,16 @@ <h2 class="govuk-heading-m govuk-!-margin-top-6">How levels here could affect ne
...
+            0m
+          {% else %}
+            {{ band.level }}m
+          {% endif %}
```
Comment (nikiwycherley) [inline server/views/station.html:242]: logic should be worked out in the model and just pass through the worked out value

File: server/src/js/components/charts.js [hunk 4]
```diff
@@ -48,12 +48,14 @@ function LineChart (containerId, data) {
   const area = d3Area().curve(curveMonotoneX)
     .x((d) => { return xScale(new Date(d.ts)) })
     .y0((d) => { return height })
-    .y1((d) => { return yScale(d._) })
+    // .y1((d) => { return yScale(d._) })
```
Comment (nikiwycherley) [inline server/src/js/components/charts.js:51]: remove commented code

File: server/src/js/components/charts.js [hunk 5]
```diff
@@ -48,12 +48,14 @@ function LineChart (containerId, data) {
...
+    .y1(d => { return yScale(data.plotNegativeValues === false && d._ < 0 ? 0 : d._) }) // *DBL below zero addition
 
   // Line generator
   const line = d3Line().curve(curveMonotoneX)
     .x((d) => { return xScale(new Date(d.ts)) })
-    .y((d) => { return yScale(d._) })
+    // .y((d) => { return yScale(d._) })
```
Comment (nikiwycherley) [inline server/src/js/components/charts.js:57]: remove commented code

File: server/src/js/components/charts.js [hunk 6]
```diff
@@ -222,8 +225,19 @@ function LineChart (containerId, data) {
...
     toolTipY = pointer(e)[1]
-    toolTip.select('text').append('tspan').attr('class', 'tool-tip-text__strong').attr('dy', '0.5em').text(Number(dataPoint._).toFixed(2) + 'm')
+
+    // Set values below zero to display zero, rather than the actual value
+    // dataPoint.checkTooltipValue = Number(dataPoint._).toFixed(2)
+    // if (!window.flood.model.station.isCoastal) {
+    //   if (dataPoint.checkTooltipValue <= 0) {
+    //     dataPoint.checkTooltipValue = '0'
+    //   }
+    // }
```
Comment (nikiwycherley) [inline server/src/js/components/charts.js:235]: remove commented code

File: server/src/js/components/charts.js [hunk 7]
```diff
@@ -222,8 +225,19 @@ function LineChart (containerId, data) {
...
+    const value = data.plotNegativeValues === false && (Math.round(dataPoint._ * 100) / 100) <= 0 ? '0' : dataPoint._.toFixed(2) // *DBL below zero addition
+    // toolTip.select('text').append('tspan').attr('class', 'tool-tip-text__strong').attr('dy', '0.5em').text(dataPoint.checkTooltipValue + 'm')
+    toolTip.select('text').append('tspan').attr('class', 'tool-tip-text__strong').attr('dy', '0.5em').text(value + 'm') // *DBL below zero addition
```
Comment (nikiwycherley) [inline server/src/js/components/charts.js:238]: remove commented code

File: server/src/js/components/charts.js [hunk 8]
```diff
@@ -306,12 +320,24 @@ function LineChart (containerId, data) {
   const clipInner = svgInner.append('g').attr('clip-path', 'url(#clip)')
 
   // Add observed and forecast elements
-  let observedArea, observed, forecastArea, forecast
+  let observedArea, observed, forecastArea, forecast, correctedLines, newLine
   if (hasObserved) {
```
Comment (nikiwycherley) [inline server/src/js/components/charts.js:324]: not needed with the new changes

File: server/src/js/components/charts.js [hunk 9]
```diff
@@ -306,12 +320,24 @@ function LineChart (containerId, data) {
...
+        }
+        return newLine
+      })
+      observedArea = svg.select('.observed').append('path').datum(correctedLines).classed('observed-area', true)
+      observed = svg.select('.observed').append('path').datum(correctedLines).classed('observed-line', true)
+    } else {
+      observedArea = svg.select('.observed').append('path').datum(observedLines).classed('observed-area', true)
+      observed = svg.select('.observed').append('path').datum(observedLines).classed('observed-line', true)
+    }
   }
```
Comment (nikiwycherley) [inline server/src/js/components/charts.js:341]: not needed with the new changes

File: server/src/js/components/charts.js [hunk 10]
```diff
@@ -457,9 +483,6 @@ function LineChart (containerId, data) {
     thresholdsContainer.on('mouseover', (e) => {
       if (e.target.closest('.threshold')) hideTooltip()
     })
-  } else {
-    // no Values so hide chart div
-    document.getElementsByClassName('defra-line-chart')[0].style.display = 'none'
   }
```
Comment (nikiwycherley) [inline server/src/js/components/charts.js:486]: should be left in

File: test/models/river-and-sea-levels.js (hunk 1 above)
Comment (nikiwycherley) [reply test/models/river-and-sea-levels.js:64]: after checking you are correct should be 0m

**Review by neilbmclaughlin — DISMISSED** [review]
@cliff-defra - can you change the PR title to 'FSR-594 add below zero message' please.

This will ensure a clean message in the commit history for the PR merge to development.

**Review by nikiwycherley — APPROVED** [review]
all looks good

**Pull request merged**